narwhals/functions.py

```diff
@@ -1103,6 +1131,13 @@ def _scan_parquet_impl(
         import pyarrow.parquet as pq  # ignore-banned-import
 
         native_frame = pq.read_table(source, **kwargs)
+    elif implementation is Implementation.SQLFRAME:
+        session = kwargs.pop("session")
```

Comment:
What about having a friendlier error message (that repeats the advice in the docstring) like:

```suggestion
        session = kwargs.pop("session", None)
        if not session:
            raise ValueError("Spark like backends require a session object to be passed in `kwargs`.)
```

Reply:
You could use something like (https://github.com/narwhals-dev/narwhals/issues/2123#issuecomment-2692160476) to add an `@overload` for it.

I suppose the match might be:

```py
from __future__ import annotations

from typing import TYPE_CHECKING
from typing import Any
from typing import Literal
from typing import TypedDict
from typing import overload

from narwhals.utils import Implementation

if TYPE_CHECKING:
    from types import ModuleType

    from typing_extensions import Never
    from typing_extensions import TypeAlias

    from narwhals.dataframe import LazyFrame

class _SparkLikeScanParquetKwds(TypedDict):
    ... # declare whatever is required/optional

_SparkLikeBackend: TypeAlias = Literal[Implementation.PYSPARK, Implementation.SQLFRAME, "pyspark", "sqlframe"]


@overload
def scan_parquet(
    source: str,
    *,
    backend: None = ...,
    native_namespace: None = ...,
    **kwargs: Any,
) -> Never: ...
@overload
def scan_parquet(
    source: str,
    *,
    backend: _SparkLikeBackend,
    native_namespace: None = ...,  # noqa: ARG001
    **kwargs: Unpack[_SparkLikeScanParquetKwds],
) -> LazyFrame[Any]: ...
@overload
def scan_parquet(
    source: str,
    *,
    backend: ModuleType | Implementation | str | None = None, # <------- Probably needs to be narrowed
    native_namespace: ModuleType | None = None,  # noqa: ARG001
    **kwargs: Any,
) -> LazyFrame[Any]: ...
def scan_parquet(
    source: str,
    *,
    backend: ModuleType | Implementation | str | None = None,
    native_namespace: ModuleType | None = None,  # noqa: ARG001
    **kwargs: Any,
) -> LazyFrame[Any]:
```

Reply:
Important thing here is `ModuleType` can't appear in the more specific `@overload` definitions (I think) - to avoid an `[overlapping-overload]`